pysph/sph/integrator.py

```diff
@@ -90,11 +90,18 @@ def _get_explicit_dt_adapt(self):
             for pa in a_eval.particle_arrays:
                 if 'dt_adapt' in pa.properties:
                     if pa.gpu is not None:
-                        from compyle.array import minimum
-                        min_val = minimum(pa.gpu.dt_adapt)
+                        if pa.gpu.get_number_of_particles() > 0:
+                            from compyle.array import minimum
+                            min_val = minimum(pa.gpu.dt_adapt)
+                        else:
+                            min_val = 1e20
```

Comment:
I think this value and the `dt_min` value above should be set to `np.inf` instead of this arbitrary 1e20 number.

Reply:
done